### ISSUE TITLE
Update to docutils 0.16

### ISSUE THREAD

**Issue by domdfcoding** [tk0miya/docutils-stubs#33]
Hi,

Would it be possible to have an updated release that supports docutils 0.16?

From what I can tell the changes from 0.14 to 0.16 are:

* adding `languages/ko.py` and `parsers/rst/languages/ko.py`
* adding a `width` option to the `Table` directive
* `Inliner.phrase_ref` now has a default value for `text` of `None`
* `TitlePromoter.promote_title` now returns `bool` not `Optional[int]`. Same for `promote_subtitle`
* `StripClassesAndElements.check_classes` now returns `bool`, not `int`
* `utils.unescape` moved to `nodes`
* `utils.unique_combinations` was deprecated, although the return type might be the same.

I dare say the current stubs will work with 0.16, but pip refuses to install them as the docutils requirement is pinned to 0.14 but one of my dependencies requires >=0.16 .

I'd be happy to submit a PR with these changes.

**Comment by tk0miya** [comment]
Contributions are always welcome :-)

**Comment by gaborbernat** [comment]
I think now we need to bump to https://pypi.org/project/docutils/0.17; though any reason why we hard pin the docutils version? Pythons gradual typing would mean we'd still be able to use it for newer versions, just the new parts would not be type annotated.

**Comment by tk0miya** [comment]
> In addition, stub-only distributions SHOULD indicate which version(s) of the runtime package are supported by indicating the runtime distribution's version(s) through normal dependency data. 

PEP-561 recommends to indicate the target version of the stub package. So we've pinned it. Is this wrong way?

**Comment by gaborbernat** [comment]
I think you want ``>=`` rather than ``==`` 🤔

**Comment by Huy-Ngo** [comment]
Maybe it's `==` because the stubs is supposed to cover all attributes/methods available?
If that's the case then I'm not sure we can just jump from 0.14 to 0.16 and skipping 0.15.

**Comment by McSinyx** [comment]
> I'm not sure we can just jump from 0.14 to 0.16 and skipping 0.15

We shouldn't, there should be a release to match each docutils release that we want to support.  Note that the current CI doesn't perform integration with docutils' source.

@tk0miya, do you want some help maintaining these stubs?  I know that you are working really actively on (and thus is busy with) Sphinx so you probably do not have much time and energy left for less important auxiliary packages like this one.  I have quite some experience with Python static typing and I am working (with @Huy-Ngo) on [a downstream repo](https://man.sr.ht/~cnx/ipwhl) where docutils-stubs is a common (and conflicting) dependency for many popular packages.

**Comment by mborsetti** [comment]
> I think you want `>=` rather than `==` 🤔

Correct, as `==` does not scale as a single package that isn't updated can cascade a chain of non-upgradable packages.
